pkg/virtctl/expose/expose_test.go

```diff
@@ -2,367 +2,441 @@ package expose_test
-				Expect(obtainedService).ToNot(BeNil())
-				Expect(obtainedService.Spec.IPFamilies).Should(HaveLen(2))
-				Expect(obtainedService.Spec.IPFamilies[0]).To(Equal(k8sv1.IPv6Protocol))
-				Expect(obtainedService.Spec.IPFamilies[1]).To(Equal(k8sv1.IPv4Protocol))
-			})
+		BeforeEach(func() {
+			vmi = libvmi.New(libvmi.WithLabel(labelKey, labelValue))
+			vmi, err := virtClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
-			It("should succeed with no IPFamily", func() {
-				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
-					"--port", "9999", "--target-port", "http")
-				Expect(cmd()).To(Succeed(), "should succeed when no IP family is provided")
-				Expect(obtainedService).ToNot(BeNil())
-				Expect(obtainedService.Spec.IPFamilies).To(BeEmpty())
-			})
+			vm, err = virtClient.KubevirtV1().VirtualMachines(metav1.NamespaceDefault).Create(context.Background(), libvmi.NewVirtualMachine(vmi), metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
 
-			It("should fail with an invalid IPFamily", func() {
-				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
-					"--port", "9999", "--target-port", "http", "--ip-family", "ipv14")
-				Expect(cmd()).To(HaveOccurred(), "should fail on an invalid IP family")
+			vmirs = kubecli.NewMinimalVirtualMachineInstanceReplicaSet("vmirs")
+			vmirs.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
+			vmirs.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labelKey: labelValue,
+				},
+			}
+			vmirs, err = virtClient.KubevirtV1().VirtualMachineInstanceReplicaSets(metav1.NamespaceDefault).Create(context.Background(), vmirs, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		DescribeTable("creating a service with default settings", func(resType string) {
+			err := runCommand(resType, getResName(resType), "--name", serviceName, "--port", servicePortStr)
+			Expect(err).ToNot(HaveOccurred())
+
+			service, err := kubeClient.CoreV1().Services(metav1.NamespaceDefault).Get(context.Background(), serviceName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(service.Spec.Selector).To(HaveLen(1))
+			Expect(service.Spec.Selector).To(HaveKeyWithValue(labelKey, labelValue))
+			Expect(service.Spec.Ports).To(HaveLen(1))
+			Expect(service.Spec.Ports[0].Port).To(Equal(servicePort))
+			Expect(service.Spec.Ports[0].Protocol).To(Equal(k8sv1.ProtocolTCP))
+			Expect(service.Spec.ClusterIP).To(BeEmpty())
+			Expect(service.Spec.Type).To(Equal(k8sv1.ServiceTypeClusterIP))
+			Expect(service.Spec.IPFamilies).To(BeEmpty())
+			Expect(service.Spec.ExternalIPs).To(BeEmpty())
+			Expect(service.Spec.IPFamilyPolicy).To(BeNil())
```

Comment:
Is it possible to make use of Equality matcher, here and in other tests? (create the expected object and compare with actual)
Making use of the Equality matcher makes it easier to realize the expected result and reduce potential for unexpected results (in case some other fileds are changed where they shouldn't)

Reply:
At least in the case of Ports there will be additional fields, which we are not interested in. But you have a point.